src/traits.rs

```diff
@@ -137,3 +137,33 @@ pub trait Concatenate {
     /// if the matrices can not be concatenated due to mismatching dimensions
     fn concat_horizontal(self, other: Self) -> Result<Self::Output, MathError>;
 }
+
+/// Is implemented by [`Z`](crate::integer::Z) instances to calculate the `gcd`
+pub trait Gcd<T = Self> {
+    type Output;
+
+    /// Outputs the greatest common divisor (gcd) of the two given values
+    /// with `gcd(a,0) = |a|`.
+    ///
+    /// Paramters:
+    /// - `other`: specifies one of the values of which the gcd is computed
+    ///
+    /// Returns the greatest common divisor of `self` and `other`.
+    fn gcd(&self, other: T) -> Self::Output;
+}
+
+/// Is implemented by [`Z`](crate::integer::Z) instances to calculate the
+/// extended `gcd`
+pub trait Xgcd<T = Self> {
+    type Output;
+
+    /// Outputs the extended greatest common divisor (xgcd) of the two given values,
+    /// i.e. a triple `(gcd(a,b), x, y)`, where `a*x + b*y = gcd(a,b)*`.
+    ///
+    /// Paramters:
+    /// - `other`: specifies one of the values of which the gcd is computed
+    ///
+    /// Returns a triple `(gcd(a,b), x, y)` containing the greatest common divisor,
+    /// `x`, and `y` s.t. `gcd(a,b) = a*x + b*y`.
+    fn xgcd(&self, other: T) -> Self::Output;
+}
```

Comment:
On could consider combining this trait, since gcd and xgcd will probably not be implemented on their own, hence one can consider implementing them as a single trait with both functionalities, but I am fine with both. Feel free to do as you like here

Reply:
Yes, I've thought about this option too. I chose to do it the separate way, because Flint also offers a `gcd` computation for `Q` (explicitely defined in the documentation), but does not offer `xgcd` computation for `Q`. This way, if we would ever want to take advantage of `Q::gcd`, the modularity of the trade would come in handy.